### PR TITLE
Issue 1820 - Add timeout option for the nonblocking calls

### PR DESCRIPTION
**Pull request by zabooma**
Add timeout option for the nonblocking calls. 
Add wait/16 for the blocking calls, as suggested by Consul doc
Cleanup unused OptionMap initialization in ConsulClientImpl constructor and import.

Issue #1820

### CODE REVIEW

**Comment by miklish** [comment]
We will be modifying the PR as well as [1810](https://github.com/networknt/light-4j/pull/1810) to streamline use of ConsulConfig values (in particular, how default values are handled and how string to integer/long conversions are done for time values)

**Comment by miklish** [comment]
We will be modifying this PR to streamline use of ConsulConfig values (in particular, how default values are handled and how string to integer/long conversions are done for time values)

**Comment by miklish** [comment]
@zabooma  - Please update to add and use new method ConsulUtils.getNonBlockingWaitInSecond()

File: consul/src/main/resources/config/consul.yml [hunk 1]
```diff
@@ -26,6 +26,8 @@ ttlCheck: true
 wait: 600s
 # Additional buffer of time to allow Consul to terminate the blocking query connection
 timeoutBuffer: 5s
+# Response timeout for the non blocking calls
+nonBlockingWait: 5s
```
Comment (miklish) [inline consul/src/main/resources/config/consul.yml:30]: This looks like it is a redundant property. We should consider using requestTimeout in this same consul.yml... It sidesteps all issues of String conversion, and accessor for it already exists in ConsulConfig

Comment (zabooma) [reply consul/src/main/resources/config/consul.yml]: Code is changed and it uses requestTimeout instead. Part of this PR

**Review by stevehu — APPROVED** [review]

**Comment by stevehu** [comment]
Sorry. I just realized that this PR has other changes. It seems we still have the decryption issue after fixing the pom.xml

**Comment by zabooma** [comment]
@stevehu I am able to build 1.6.x locally with no issues. powermock dependencies in config now include powermock which was also added as part of this PR.

**Comment by AkashWorkGit** [comment]
FYI:The Junit could not be created for the changes in consulCientImpl class after several attempts but the manual unit test & functional testing was completed successfully.

**Comment by zabooma** [comment]
@stevehu something is wrong with 1.6.x branch. Few other people on team also see this error. The issue is that for some reason if you do git pull of 1.6.x from an earlier clone it does not pull the change to pom. However if you clone into a new location and do build there it builds with no issues.

You can see here that powermock is actually in 1.6.x branch: 
https://github.com/networknt/light-4j/blob/1.6.x/config/pom.xml

This is a build from the fresh clone:

`[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Parent POM 1.6.46-SNAPSHOT:
[INFO] 
[INFO] Parent POM ......................................... SUCCESS [  4.392 s]
[INFO] decryptor .......................................... SUCCESS [ 12.487 s]
[INFO] config ............................................. SUCCESS [ 21.467 s]
[INFO] utility ............................................ SUCCESS [ 14.513 s]
[INFO] http-string ........................................ SUCCESS [  5.278 s]
[INFO] http-url ........................................... SUCCESS [ 12.881 s]
[INFO] service ............................................ SUCCESS [ 12.129 s]
[INFO] status ............................................. SUCCESS [ 14.657 s]
[INFO] common ............................................. SUCCESS [ 11.017 s]
[INFO] handler ............................................ SUCCESS [ 12.326 s]
[INFO] monad-result ....................................... SUCCESS [  9.913 s]
[INFO] switcher ........................................... SUCCESS [  9.298 s]
[INFO] registry ........................................... SUCCESS [ 15.355 s]
[INFO] balance ............................................ SUCCESS [ 10.329 s]
[INFO] cluster ............................................ SUCCESS [ 13.089 s]
[INFO] client ............................................. SUCCESS [01:26 min]
[INFO] exception .......................................... SUCCESS [ 11.673 s]
[INFO] security ........................................... SUCCESS [ 14.942 s]
[INFO] correlation ........................................ SUCCESS [ 10.971 s]
[INFO] mask ............................................... SUCCESS [ 10.162 s]
[INFO] body ............................................... SUCCESS [ 20.412 s]
[INFO] audit .............................................. SUCCESS [ 36.795 s]
[INFO] dump ............................................... SUCCESS [ 13.672 s]
[INFO] info ............................................... SUCCESS [ 12.339 s]
[INFO] health ............................................. SUCCESS [ 11.148 s]
[INFO] header ............................................. SUCCESS [ 10.869 s]
[INFO] consul ............................................. SUCCESS [ 23.840 s]
[INFO] zookeeper .......................................... SUCCESS [ 14.270 s]
[INFO] server ............................................. SUCCESS [ 16.722 s]
[INFO] metrics ............................................ SUCCESS [ 32.768 s]
[INFO] prometheus ......................................... SUCCESS [ 11.636 s]
[INFO] sanitizer .......................................... SUCCESS [ 11.621 s]
[INFO] traceability ....................................... SUCCESS [ 10.808 s]
[INFO] cors ............................................... SUCCESS [ 12.446 s]
[INFO] rate-limit ......................................... SUCCESS [ 11.279 s]
[INFO] email-sender ....................................... SUCCESS [  7.953 s]
[INFO] resource ........................................... SUCCESS [ 10.776 s]
[INFO] content ............................................ SUCCESS [ 11.456 s]
[INFO] data-source ........................................ SUCCESS [ 11.423 s]
[INFO] basic-auth ......................................... SUCCESS [ 12.896 s]
[INFO] deref-token ........................................ SUCCESS [ 11.203 s]
[INFO] ip-whitelist ....................................... SUCCESS [ 12.273 s]
[INFO] encode-decode ...................................... SUCCESS [ 12.691 s]
[INFO] logger-config ...................................... SUCCESS [  4.274 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  11:02 min
[INFO] Finished at: 2023-10-30T11:52:58-04:00
[INFO] ------------------------------------------------------------------------
`

**Comment by stevehu** [comment]
I tried to clone it and switch to the 1.6.x branch on my local minutes ago. It failed in the config module. So, a fresh checkout doesn't work for me.

**Comment by miklish** [comment]
@stevehu 
What system are you building this on? It builds on all of our macs... We see issues on Windows however. Have not tried on Linux.

However, how was this PR able to get merged if the build failed? So, I assume that the build is working in your automated pre-merge build on GitHub?

**Comment by miklish** [comment]
@stevehu 
Have you tried building after the PR update that includes the PowerMock dependencies? We believe that update should resolve unit test failures in the config module

File: config/pom.xml [hunk 2]
```diff
@@ -65,6 +65,14 @@
...
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-module-junit4</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-api-mockito2</artifactId>
+        </dependency>
```
Comment (miklish) [inline config/pom.xml:75]: You can see in this PR an update was made to include new PowerMock dependencies in pom.xml

**Comment by stevehu** [comment]
This is the error I have encountered. 

```
[INFO] Running com.networknt.config.JsonMapperTest
s = {"Integer":111,"string":"hello","Long":111,"long":111,"int":111}
{Integer=111, string=hello, Long=111, long=111, int=111}
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 s - in com.networknt.config.JsonMapperTest
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ConfigDecryptTest.testDecryptMap:43 expected:<[password]> but was:<[CRYPT:08eXg9TmK604+w06RaBlsPQbplU1F1Ez5pkBO/hNr8w=]>
[ERROR]   ConfigDecryptTest.testDecryptObject:69 expected:<[password]> but was:<[CRYPT:08eXg9TmK604+w06RaBlsPQbplU1F1Ez5pkBO/hNr8w=]>
[INFO] 
[ERROR] Tests run: 58, Failures: 2, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Parent POM 1.6.46-SNAPSHOT:
[INFO] 
[INFO] Parent POM ......................................... SUCCESS [  0.601 s]
[INFO] decryptor .......................................... SUCCESS [  3.267 s]
[INFO] config ............................................. FAILURE [  5.951 s]

```

**Comment by miklish** [comment]
Thanks @stevehu - we are looking into this specific error now... we will either provide an update or rollback our new unit test. We will provide an update here asap

**Comment by miklish** [comment]
Hi Steve, we believe we have fixed this issue.

Please see PR https://github.com/networknt/light-4j/pull/1941

**Comment by stevehu** [comment]
I have merged that PR, and the build is OK on my desktop. Thanks a lot for the quick response.

**Pull request merged**